CMakeLists.txt

```diff
@@ -91,28 +91,40 @@ include(PreventInBuildInstalls)
 
 # Check if recursive flag exists
 include(CheckFortranCompilerFlag)
-check_fortran_compiler_flag("-recursive" _recursiveFlag)
-check_fortran_compiler_flag("-frecursive" _frecursiveFlag)
-check_fortran_compiler_flag("-Mrecursive" _MrecursiveFlag)
+if(CMAKE_Fortran_COMPILER_ID STREQUAL Flang)
+  check_fortran_compiler_flag("-Mrecursive" _MrecursiveFlag)
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL GNU)
+  check_fortran_compiler_flag("-frecursive" _frecursiveFlag)
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL Intel)
+  check_fortran_compiler_flag("-recursive" _recursiveFlag)
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL XL)
+  check_fortran_compiler_flag("-qrecur" _qrecurFlag)
+endif()
```

Comment:
I think it is a good idea to have a non-empty `else()` for other compiler IDs. I would considers using either:
```cmake
else()
  check_fortran_compiler_flag("-recursive" _recursiveFlag)
  check_fortran_compiler_flag("-frecursive" _frecursiveFlag)
  check_fortran_compiler_flag("-Mrecursive" _MrecursiveFlag)
  check_fortran_compiler_flag("-qrecur" _qrecurFlag)
endif()
```
or:
```cmake
else()
  message( WARNING "Fortran local arrays should be allocated on the stack."
    " Please assure your compiler guarantees that."
    " See https://github.com/Reference-LAPACK/lapack/pull/188 and references therein." )
endif()
```